@utahdts/utah-design-system/react/components/forms/FileInput/FileInput.jsx

```diff
@@ -0,0 +1,177 @@
+import { useImmer } from 'use-immer';
+import {
+  useCallback, useEffect, useRef
+} from 'react';
+import { isEqual, isFunction } from 'lodash';
+import { RequiredStar } from '../RequiredStar';
+import { joinClassNames } from '../../../util/joinClassNames';
+import { Tag } from '../../buttons/Tag';
+import { useAriaMessaging } from '../../../contexts/UtahDesignSystemContext/hooks/useAriaMessaging';
+import { ErrorMessage } from '../ErrorMessage';
+import { FileContext } from './context/FileContext';
+
+/**
+ * @param {object} props
+ * @param {import('react').ReactNode} [props.children]
+ * @param {string} [props.acceptedFileTypes]
+ * @param {string} [props.className]
+ * @param {string} [props.errorMessage]
+ * @param {string} [props.hint]
+ * @param {string} props.id
+ * @param {import('react').Ref<HTMLDivElement>} [props.innerRef]
+ * @param {string} props.label
+ * @param {boolean} [props.multiple]
+ * @param {string} [props.name]
+ * @param {boolean} [props.isDisabled]
+ * @param {boolean} [props.isRequired]
+ * @param {(files: FileList, event: import('react').ChangeEvent<HTMLInputElement>) => void} [props.onChange]
+ * @param {FileList} [props.value]
+ * @returns {import('react').JSX.Element}
+ */
+
+export function FileInput({
+  acceptedFileTypes,
+  children,
+  className,
+  errorMessage,
+  hint,
+  id,
+  innerRef,
+  isDisabled,
+  isRequired,
+  label,
+  multiple,
+  name,
+  onChange,
+  value,
+}) {
+  const { addPoliteMessage } = useAriaMessaging();
+  const lf = new Intl.ListFormat('en');
+  const [isDragged, setDragged] = useImmer(false);
+  const [files, setFiles] = useImmer(value || []);
+  const inputRef = useRef(null);
+
+  const currentOnChange = useCallback((/** @type {import('react').ChangeEvent<HTMLInputElement>} */ event) => {
+    // @ts-ignore
```

Comment:
Interestingly, the @ts-ignore is hiding the fact that inputRef.current may be null/undefined and if it is then this code will throw an error. It appears that it should be `inputRef.current?.files`. And then if that's the case, then it is possible to pass null to onChange so the type at the top should be .
```
 * @param {(files: FileList | null, event: import('react').ChangeEvent<HTMLInputElement>) => void} [props.onChange]
```
It still gets the error `Property 'files' does not exist on type 'never'.ts(2339)`. But this is because the ref doesn't have a type:
`  const inputRef = useRef(null);`
should be
`  const inputRef = useRef(/** @type {| null} */ (null));`
Then the @ts-ignore is not needed.

The next @ts-ignore can also be removed if it's changed to:
```    setFiles(inputRef.current.files ?? []);```

Reply:
`const inputRef = useRef(/** @type {HTMLInputElement | null} */ (null));`